lib/std/ioutils.nim

```diff
@@ -0,0 +1,50 @@
+when defined(windows):
+  proc c_dup(oldfd: FileHandle): FileHandle {.
+    importc:"_dup", header: "<io.h>".}
+  proc c_dup2(oldfd: FileHandle, newfd: FileHandle): cint {.
+    importc: "_dup2", header: "<io.h>".}
+else:
+  proc c_dup(oldfd: FileHandle): FileHandle{.
+    importc: "dup", header: "<unistd.h>".}
+  proc c_dup2(oldfd: FileHandle, newfd: FileHandle): cint {.
+    importc: "dup2", header: "<unistd.h>".}
+
+when false:
+  const SupportIoctlInheritCtl = (defined(linux) or defined(bsd)) and
+                                not defined(nimscript)
+  when SupportIoctlInheritCtl:
+    var
+      FIOCLEX {.importc, header: "<sys/ioctl.h>".}: cint
+      FIONCLEX {.importc, header: "<sys/ioctl.h>".}: cint
+
+## Also defined in std/posix and system/io
+proc strerror(errnum: cint): cstring {.importc, header: "<string.h>".}
+when not defined(NimScript):
+  var errno {.importc, header: "<errno.h>".}: cint ## error variable
+
+template checkError(ret: cint)=
+  if ret == -1:
+    when not defined(NimScript):
+      raise newException(IOError, $strerror(errno))
+    else:
+      quit(1)
```

Comment:
I know quit(1) is used in some other related places but maybe `doAssert false` is better refs https://forum.nim-lang.org/t/4042